### PR TITLE
fix(Core/Quest): Death Comes From On High

### LINKED ISSUE #2607: [DK-Quest #12641] - Death Comes From On High
[DK-Quest #12641] - Death Comes From On High

##### SMALL DESCRIPTION:
* After taking control of the eye of Acherus, the eye of Acherus automatically goes to New Avalon as planned, but due to the very slow manner, it is also possible to cast spells with the eye before reaching the destination.

* Once arrived in New Avalon, it is not possible to go up or down with the eye immediately, it is necessary to advance a little before having this possibility.
* Even after control of eye movement is possible, it remains slower than it should be.
* NPCs don't attack the eye unless they get very close.

##### EXPECTED BLIZZLIKE BEHAVIOUR:

https://www.youtube.com/watch?v=9V8KWMMa0aU

https://www.wowhead.com/quest=12641/death-comes-from-on-high
https://wowgaming.altervista.org/aowow/?quest=12641

##### BRANCH(ES):
master

##### AC HASH/COMMIT:
https://github.com/azerothcore/azerothcore-wotlk/commit/9f690a37412c521fd0cd7a333d4cc0b554ccc905

##### OPERATING SYSTEM:
Ubuntu 18.04

##### MODULES:
- mod_eluna
- mod_transmo

##### OTHER CUSTOMIZATIONS:
No custom

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/87446245-dk-quest-12641-death-comes-from-on-high?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by imxari**
fix(CORE/Quest): Death Comes From On High

## CHANGES PROPOSED:
-  Tweaked the method of transporting the Eye of Acherus from Ebon Hold to New Avalon(This helps resolve the player gaining control too early).
-  Added a few lines to change the flight speed of the Eye of Acherus as there doesn't appear to be a way to do this in the database.

## ISSUES ADDRESSED:
- Closes #2607 

## TESTS PERFORMED:
- Building the core (Builds without errors)
- Completing the quest in-game (Speed is more BlizzLike both during the flight to New Avalon and after)

## HOW TO TEST THE CHANGES:

 - Build the core
 - Login and create a Death Knight
 - Do the quest "Death Comes From On High"

## KNOWN ISSUES AND TODO LIST:

- [x] The Scarlet Crusade NPCs at New Avalon don't aggro on the Eye of Acherus unless you get incredibly close. I couldn't find a reason for it.

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp [hunk 1]
```diff
@@ -29,6 +29,9 @@ enum eyeOfAcherus
     EYE_TEXT_LAUNCH                 = 0,
     EYE_TEXT_CONTROL                = 1,
 
+    EYE_POINT_DESTINATION_1	    = 0,
```
Comment (Viste) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp:32]: ident broken

Comment (h0lysp4nk) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp]: Fixed.

**Comment by Kitzunu** [comment]
>The Scarlet Crusade NPCs at New Avalon don't aggro on the Eye of Acherus unless you get incredibly close. I couldn't find a reason for it.
@h0lysp4nk 
Perhaps this could be done in SAI?

**Comment by h0lysp4nk** [comment]
@Kitzunu Yeah that can definitely be done in SAI, I'll take a look at doing that. If I remember correctly C++ scripts override SAI so the things done in the C++ script will need to be moved over to SAI. That should be a rather easy change. I'll work on doing that now :)

**Comment by h0lysp4nk** [comment]
@Kitzunu Done, added a SAI to the Scarlet Crusaders to make them aggro on the Eye of Acherus. Looks to work well :)

File: data/sql/updates/pending_db_world/rev_1587740947996468714.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1587740947996468714');
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES (28529, 0, 2, 0, 1, 0, 100, 0, 0, 0, 1000, 3000, 0, 49, 0, 0, 0, 0, 0, 0, 9, 28511, 1, 25, 1, 0, 0, 0, 0, 'Scarlet Crusader - Out of Combat - Start Attacking');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1587740947996468714.sql:2]:
```suggestion
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES 
(28529, 0, 2, 0, 1, 0, 100, 0, 0, 0, 1000, 3000, 0, 49, 0, 0, 0, 0, 0, 0, 9, 28511, 1, 25, 1, 0, 0, 0, 0, 'Scarlet Crusader - Out of Combat - Start Attacking');
```

File: data/sql/updates/pending_db_world/rev_1587740947996468714.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1587740947996468714');
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1587740947996468714.sql:2]: always use `DELETE` before `INSERT`

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1587740947996468714.sql:2]: Oh how did I miss that

Comment (h0lysp4nk) [reply data/sql/updates/pending_db_world/rev_1587740947996468714.sql:2]: Good point, I'll add that now.

Comment (h0lysp4nk) [reply data/sql/updates/pending_db_world/rev_1587740947996468714.sql:4]: Done

File: data/sql/updates/pending_db_world/rev_1587740947996468714.sql [hunk 4]
```diff
@@ -1,3 +1,5 @@
 INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1587740947996468714');
+
+DELETE FROM `smart_scripts` WHERE entryorguid=28529 AND id=2;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1587740947996468714.sql:3]:
```suggestion
DELETE FROM `smart_scripts` WHERE `entryorguid`=28529 AND `id`=2;
```

**Comment by BarbzYHOOL** [comment]
Thank you for that PR!

**Comment by h0lysp4nk** [comment]
Mm, it's stating that @Kitzunu has requested a change although I don't see it?

**Comment by Kitzunu** [comment]
Don't worry you already added the change, it is because I haven't had time to approve yet.

**Comment by h0lysp4nk** [comment]
Aha okay no worries, thanks for letting me know. Thought I was missing something :)

**Comment by wowissues** [comment]
> Aha okay no worries, thanks for letting me know. Thought I was missing something :)

Please tell me, is this PR correcting the speed of movement of the eye?

**Comment by felwyn** [comment]
Works good now! Thanks to all involved! @FrancescoBorzi

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**